R/distToCells.R

```diff
@@ -50,86 +52,85 @@
-  
+distToCells <- function (object,
+                        x_cells,
+                        img_id,
+                        name = "distToCells",
+                        coords = c("Pos_X","Pos_Y"),
+                        metric="min",
+                        return_neg = TRUE,
+                        BPPARAM = SerialParam()){
+  .valid.distToCells.input(object, x_cells, name, coords, metric,
+                              img_id, return_neg)
   cur_meta <- metadata(object)
   metadata(object) <- list()
-  
   cur_intmeta <- int_metadata(object)
-  
   object$x_cells <- x_cells
-
   cur_out <- bplapply(
     unique(colData(object)[[img_id]]),
-    function(x){
-      
-      cur_obj <- object[,as.character(colData(object)[[img_id]]) == x]
-      
+    function(x) {
+      cur_obj <- object[, as.character(colData(object)[[img_id]]) == x]
       cur_obj[[name]] <- NA
-      if (sum(cur_obj$x_cells) == 0 | sum(cur_obj$x_cells) == ncol(cur_obj)) {
+      if (sum(cur_obj$x_cells) == 0 | sum(cur_obj$x_cells) ==
+          ncol(cur_obj)) {
         return(cur_obj)
       }
-      
       patch_cells <- which(cur_obj$x_cells)
       non_patch_cells <- which(!cur_obj$x_cells)
-      
       if (is(object, "SpatialExperiment")) {
         dist_mat <- distances(spatialCoords(cur_obj))
-      } else {
-        dist_mat <- distances(as.matrix(colData(cur_obj)[,coords]))
       }
-      
-      pos_dist <- distance_columns(dist_mat,column_indices = patch_cells)
-      dist_to_patch <- rowMins(pos_dist)
-      neg_dist <- distance_columns(dist_mat,column_indices = non_patch_cells)
-      dist_from_patch <- rowMins(neg_dist)
-      
-      # cells that had a 0 distance to the cells of interest can be substitutes
-      # with the negative distances from the cells of interest
-      if(return_neg == TRUE) {
-        dist_to_patch[dist_to_patch == 0] <- -dist_from_patch[dist_to_patch == 0]
+      else {
+        dist_mat <- distances::distances(as.matrix(colData(cur_obj)[,
+                                                                    coords]))
+      }
+      pos_dist <- distance_columns(dist_mat, column_indices = patch_cells)
+      neg_dist <- distance_columns(dist_mat, column_indices = non_patch_cells)
+      dist_to_patch <- switch(metric,
+                              mean = rowMeans(pos_dist),
+                              median = rowMedians(pos_dist),
+                              min = rowMins(pos_dist),
+                              max = rowMaxs(pos_dist))
+
+
+      if (return_neg == TRUE) {
+        dist_from_patch <- switch(metric,
+                                  mean = rowMeans(neg_dist),
+                                  median = rowMedians(neg_dist),
+                                  min = rowMins(neg_dist),
+                                  max = rowMaxs(neg_dist))
+
+        dist_to_patch[cur_obj$x_cells] <- -dist_from_patch[cur_obj$x_cells]
       }
       cur_obj[[name]] <- dist_to_patch
```

Comment:
as we discussed, potentially worthwhile to alter the new colData `name` entry depending on the metric chosen, so the user does not overwrite distToCells straight away with a new metric (e.g. if user chooses `max`, the entry could be called `maxDistToCells`)